apps/website/src/components/menu/menu.tsx

```diff
@@ -10,65 +18,77 @@ type Props = {
 
 export const Menu = component$<Props>(({ onClose$ }) => {
   const appState = useContext(APP_STATE);
-  const menu = [
-    {
-      label: 'Accordion',
-      path: `/docs/${appState.theme.toLowerCase()}/accordion`,
-    },
-    {
-      label: 'Alert',
-      path: `/docs/${appState.theme.toLowerCase()}/alert`,
-    },
-    {
-      label: 'Badge',
-      path: `/docs/${appState.theme.toLowerCase()}/badge`,
-    },
-    {
-      label: 'Breadcrumb',
-      path: `/docs/${appState.theme.toLowerCase()}/breadcrumb`,
-    },
-    { label: 'Button', path: `/docs/${appState.theme.toLowerCase()}/button` },
-    {
-      label: 'ButtonGroup',
-      path: `/docs/${appState.theme.toLowerCase()}/button-group`,
-    },
-    { label: 'Card', path: `/docs/${appState.theme.toLowerCase()}/card` },
-    {
-      label: 'Collapse',
-      path: `/docs/${appState.theme.toLowerCase()}/collapse`,
-    },
-    {
-      label: 'Navigation Bar',
-      path: `/docs/${appState.theme.toLowerCase()}/navigation-bar`,
-    },
-    { label: 'Drawer', path: `/docs/${appState.theme.toLowerCase()}/drawer` },
-    { label: 'Input Phone', path: `/docs/headless/input-phone` },
-    { label: 'Rating', path: `/docs/${appState.theme.toLowerCase()}/rating` },
-    { label: 'Radio', path: `/docs/${appState.theme.toLowerCase()}/radio` },
-    { label: 'Popover', path: `/docs/${appState.theme.toLowerCase()}/popover` },
-    { label: 'Select', path: `/docs/${appState.theme.toLowerCase()}/select` },
-    {
-      label: 'Spinner',
-      path: `/docs/${appState.theme.toLowerCase()}/spinner`,
-    },
-    { label: 'Tabs', path: `/docs/${appState.theme.toLowerCase()}/tabs` },
-    { label: 'Toast', path: `/docs/${appState.theme.toLowerCase()}/toast` },
-    { label: 'Toggle', path: `/docs/${appState.theme.toLowerCase()}/toggle` },
-    { label: 'Tooltip', path: `/docs/${appState.theme.toLowerCase()}/tooltip` },
-    { label: 'Slider', path: `/docs/${appState.theme.toLowerCase()}/slider` },
-    {
-      label: 'Pagination',
-      path: `/docs/${appState.theme.toLowerCase()}/pagination`,
-    },
-    {
-      label: 'Checkbox',
-      path: `/docs/${appState.theme.toLowerCase()}/checkbox`,
-    },
-    {
-      label: 'Progress',
-      path: `/docs/${appState.theme.toLowerCase()}/progress`,
-    },
-  ];
+  const menu = useStore<{ value: AppPath[] }>({
```

Comment:
I don't understand why you put the menu in a state. Is there any reason?

Reply:
In that solution, I utilized state to update the menu directly on the client after theme was set. 

But your question made me think and I revisited my approach and made some improvements in the next commit. I noticed that the server-side state for the theme was always "NOT_DEFINED," so I decided to update the theme state on the server using useLocation instead of waiting for a client-side state update. Additionally, I prevented  flickering menu which was caused by the state on every route change being defaulted to "NOT_DEFINED" 

With these changes, I believe that the new solution I have implemented should effectively address the issue at hand. Please let me know if you have any further suggestions or feedback.